### PR TITLE
Create PrecisionPlugin for megatron_ckpt_to_nemo.py trainer

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: examples/nlp/language_modeling/megatron_ckpt_to_nemo.py [hunk 1]
```diff
@@ -92,6 +97,14 @@ def get_args():
...
     parser.add_argument("--local_rank", type=int, required=False, default=os.getenv('LOCAL_RANK', -1))
     parser.add_argument("--bcp", action="store_true", help="Whether on BCP platform")
+    parser.add_argument(
+        "--precision",
+        type=str,
+        required=False,
+        default='16-mixed',
+        choices=['16-mixed', 'bf16-mixed'],
+        help="Precision value for the trainer that matches with precision of the ckpt",
+    )
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_ckpt_to_nemo.py:107]: shouldn't 32 be here as well?

**Review by yidong72 — DISMISSED** [review]
LGTM. agree with @eric about includeing the fp32 in the precision argument.

File: examples/nlp/language_modeling/megatron_ckpt_to_nemo.py (hunk 1 above)
Comment (athitten) [reply examples/nlp/language_modeling/megatron_ckpt_to_nemo.py:107]: Yes I can add that, but I was thinking we'll be just using fp16 or bf16 for training.

Comment (athitten) [reply examples/nlp/language_modeling/megatron_ckpt_to_nemo.py]: Added. Also `punctuation_capitalization_train_evaluate.py` needed `ddp_find_unused_parameters_true` to be added, added that in this PR as well.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**